src/resources/2025/PrismaticBeret.ts

```diff
@@ -243,22 +272,11 @@ const outfitPrice = (outfit: { hat: Item; pants: Item; shirt: Item }) =>
 function findOutfit(power: number, buyItems: boolean, hammerTime: boolean) {
   const { useableHats, useablePants, useableShirts } =
     getUseableClothes(buyItems);
-  const hatPowers = populateMap(
-    useableHats,
-    power,
-    have_($skill`Tao of the Terrapin`) ? 2 : 1,
-  );
-  const pantsPowers = populateMap(
-    useablePants,
-    power,
-    1 + (have_($skill`Tao of the Terrapin`) ? 1 : 0) + (hammerTime ? 3 : 0),
-  );
-  const shirtPowers = populateMap(useableShirts, power, 1);
 
-  const outfits = [...hatPowers].flatMap(([hatPower, hat]) =>
-    [...pantsPowers].flatMap(([pantsPower, pants]) =>
-      [...shirtPowers].flatMap(([shirtPower, shirt]) =>
-        hatPower + pantsPower + shirtPower === power
+  const outfits = validateItems(useableHats, power).flatMap((hat) =>
+    validateItems(useablePants, power).flatMap((pants) =>
+      validateItems(useableShirts, power).flatMap((shirt) =>
+        sumEquipmentPower({ hat, pants, shirt, hammerTime }) === power
           ? { hat, pants, shirt }
           : [],
```

Comment:
I don't think this change is necessary of valuable; you should be able to keep roughly the same logic of `populateMap` but with the `getEffectivePower` function instead of `getPower`; this change will call the `validateItems` function a tremendous number of times,